charts/karpenter/templates/aggregate-clusterrole.yaml

```diff
@@ -16,3 +16,22 @@ rules:
   - apiGroups: ["karpenter.k8s.aws"]
     resources: ["awsnodetemplates"]
     verbs: ["get", "list", "watch", "create", "delete", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "karpenter.fullname" . }}-view
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
```

Comment:
I'm a little concerned that you may want to lock-down viewing these resources. For example, the AWSNodeTemplate has free-form user-data which could include something sensitive. AFICT this would always aggregate read-only access. Maybe if it was configurable and defaults to false?